### DISCUSSION TITLE
sending a HTTP request in the background?

### DISCUSSION THREAD

**Discussion by sudomain** [TrianguloY/URLCheck#402] [Q&A] [answered]
Hello,

Is it possible to send a HTTP request in the background? Something akin to the `checkStatus` action of Automations, but also closes the dialog?

What I'm ultimately trying to do is make a GET request to a fastapi/uvicorn server running locally in Termux, but in that background/without changing the foreground Android activity. This is the URL: `http://127.0.0.1:8000/example/path`

This is what I currently have as my Automations JSON:
```
{
  "Unshort bit.ly": {
    "regex": "https?:\/\/bit\\.ly\/.*",
    "action": "unshort",
    "enabled": false
  },
  "localhost put": {
    "regex": "http:\/\/127\\.0\\.0\\.1:8000.*",
    "action": "checkStatus",
    "enabled": true
  }
}
```

Is this possible using only URLCheck? Is it possible when using URLCheck in conjunction with another app e.g. `HTTP Shortcuts`?

**Comment by TrianguloY** [comment]
Hmm, can you describe with some more details what you want to achieve? If you could provide the steps/actions that would be perfect (like "click the link", "a request is made with X", "chrome opens with url Y",...)

I'm a bit confused about the GET request, do you want to send the url to it? Or just call the endpoint and nothing else? 
And when you say "also closes the dialog" does it means you want to discard the url (like if nothing had happened) or open the browser directly?

**Reply by sudomain** [reply]
Sure, so I'm chaining together a few apps:
1. [G2L](https://f-droid.org/packages/com.easwareapps.g2l/) which takes swipe gestures as input and can "Open websites" or URLs a response to these swipes
2. URLCheck
3. Termux's FastAPI/Uvicorn Python packages to run a few personal scripts in response to GET requests. I think this is called "Webhooks" but i'm not really sure.

I want to be able to call the endpoint without changing the foreground app, which is where I think URLCheck comes in. With URLCheck's automation, I'm able to make the GET request using the checkStatus action, but I want URLCheck to close after.

**Reply by TrianguloY** [reply]
Ah, I see. That's an interesting user case, no doubt. I wonder if there is an easier way other than using a get request...have you considered tasker? 

In any case, with the current automations...I don't think you can do what you want, sorry. But I can easily add a "close" automation. Perhaps it could be useful in other situations.
It will be ready for the next version (can't say when that will be released though)

**Reply by sudomain** [reply]
I try to avoid closed-source solutions when possible. While browsing the code I noticed the upcoming webhook module. I think this may be helpful to me. I look forward to trying this and the close automation you mentioned. Thank you for the great app.

**Reply by TrianguloY** [reply, marked as answer]
It will perform a POST, not a GET (that's why I didn't mentioned it) but if that's ok for you, then perfect.

**Reply by thejeshgn** [reply]
Webhook will great :)

**Reply by sudomain** [reply]
Webhook works great. I just want to add a note for anyone else wanting to use webhook and close actions together. Currently (v3.1), to use webhook and close actions together, the close action must come before webhook action in the json. If the close action comes after the webhook action, then it will only close and not perform the webhook. For example:

```
  "Close after webhook": {
    "regex": "http:\/\/mywebsite.com.*",
    "action": "close",
    "enabled": true
  },
  "Send to webhook": {
    "regex": "http:\/\/mywebsite.com.*",
    "action": "webhook",
    "enabled": true
  }
```

I'm not sure if this is a bug

**Reply by TrianguloY** [reply]
Oh, I need to check that, I wanted the actions to run in the defined order, maybe I forgot or there is a bug 😞

Can you create an issue for it? I'll do it myself if you prefer otherwise. Thanks!
